src/composables/latestTransactionList.ts

```diff
@@ -28,14 +40,39 @@ export function useLatestTransactionList({ store }: IDefaultComposableOptions) {
 
   const tokens = computed(() => store.state.fungibleTokens.tokens);
 
+  const localPendingTransactionsList = computed(
+    (): IPendingTransaction[] => {
+      const { networkId } = store.getters.activeNetwork;
+      return (store.state.transactions as ITransactionsState).pending[networkId] || [];
+    },
+  );
+
+  function buildLatestTransaction() {
+    const transactions = [...fetchedTransactions.value];
+    pendingTransactions.value.forEach((transaction) => {
+      if (!transactions.some(({ hash }) => transaction.hash === hash)) {
+        transactions.unshift(transaction as ITransaction);
+      } else {
+        pendingTransactions.value = pendingTransactions.value.filter(
```

Comment:
This line is placed inside of a `forEach` loop which in my opinion is a bad pattern. I suggest to use more functional approach by utilising `map` & `filter` on higher level.